docs/delivery/README.md

```diff
@@ -0,0 +1,124 @@
+underlying platform (eg. RabbitMQ dead letter exchange, Amazon SQS dead letter queue, Azure Service Bus dead letter queue, etc...).
+* Be able to redirect of error'ed events from a channel.
+
+### Out of Scope
+
+* Security: No security model has been defined yet at the Knative eventing level. Native brokers might define their own security model and Knative eventing should not prevent using it.
+
+## Day 1 Proposal
+
+### Dead Letter Sink
+
+Channels are responsible for forwarding received events to subscribers. When they fail to do so, they are responsible for sending the failing events to an **dead letter sink**, potentially one per subscriber.
+
+Similarly Event Sources are responsible for sending events to a sink and when they fail to do so, they are responsible for sending the failing events to an **dead letter sink**.
+
+The dead letter sink can be a channel but it does not have to.
+
+### Dead-Letter Channel
+
+Knative Channel implementations may leverage existing platform native error handling support they might provide, like [_Dead Letter Channel_](https://www.enterpriseintegrationpatterns.com/patterns/messaging/DeadLetterChannel.html), to forward failed events from their _Dead Letter Channel_ to the configured error sink(s).
+
+### Retry
+
+Channel implementations, event sources and brokers should retry sending events before redirecting them to the dead letter sink.
+While there are many different ways to implement the retry logic
+(immediate retry, retry queue, etc...), implementations should
+rely on a common set of configuration parameters, such as
+the number of retries and the interval between retries.
+
+### Delivery Specification
+
+The goal of this delivery specification is to formally define the vocabulary related to capabilities defined above (dead letter sink, dead-letter queues and retry) to provide consistency across all Knative event sources, channel implementations and brokers.
+
+The minimal delivery specification looks like this:
+
+```go
+
+// DeliverySpec contains the delivery options for event senders,
+// such as channelable and source.
+type DeliverySpec struct {
+	// DeadLetterSink is the sink receiving events that couldn't be sent to
+	// a destination.
+	// +optional
+	DeadLetterSink *apisv1alpha1.Destination `json:"deadLetterSink,omitempty"`
+
+	// Retry is the minimum number of retries the sender should attempt when
+	// sending a event before moving it to the dead letter sink
+	// +optional
+	Retry *int32 `json:"retry,omitempty"`
+
+	// BackoffPolicy is the retry backoff policy (linear, exponential)
+	// +optional
+	BackoffPolicy *BackoffPolicyType `json:"backoffPolicy,omitempty"`
+
+	// BackoffDelay is the delay before retrying.
+	// More information on Duration format: https://www.ietf.org/rfc/rfc3339.txt
+	//
+	// For linear policy, backoff delay is the time interval between retries.
+	// For exponential policy , backoff delay is backoffDelay*10^<numberOfRetries>
```

Comment:
nit: I'm more accustomed to using base 2 for the backoff (so just double the delay each time instead of 10x it), but do not feel super strongly about this.
Thought? Do we want to cap this after say 10 times or something. This can be a todo later guided by experience.